src/DynamoCoreWpf/Views/PackageManager/PublishPackageView.xaml

```diff
@@ -558,7 +560,9 @@
                                ToolTip="{x:Static p:Resources.PublishPackageViewPackageNameTooltip}" />
                     <TextBox Name="packageNameInput"
                              Style="{StaticResource InputStyle}"
-                             Tag="{x:Static p:Resources.PublishPackageViewPackageNameWatermark}" />
+                             Tag="{x:Static p:Resources.PublishPackageViewPackageNameWatermark}"
+                             IsEnabled ="{Binding Path=CanEditName}"
+                             Text="{Binding Path=Name, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"/>
```

Comment:
@OliverEGreen Would you specify why the two-way binding is removed?

Reply:
All I can think of is unintentional omission. It seems I missed doing a lot of these bindings in one pass - sorry about this!

Reply:
@OliverEGreen Can you take a look at restoring the binding with me on this branch? So far even I added the code back, still not working for me

Reply:
Possibly a somewhat convoluted approach, I raised this PR with the fixes based off your branch: 
https://github.com/DynamoDS/Dynamo/pull/12248

Let me know if you'd rather get the changes another way and I can delete the PR :)